types/forms/index.d.ts

```diff
@@ -127,29 +133,33 @@ export interface FormFields {
     [key: string]: Field | FormFields;
 }
 
-export type FormHandleCallback = (form: Form) => void;
+export type FormHandleCallback<
+    Fields extends FormFields = FormFields,
+    Data extends (IncomingMessage | (Partial<FormData<Fields>> & { [key: string]: any })) = FormData<Fields>
```

Comment:
```suggestion
    Data extends (IncomingMessage | (Partial<FormData<Fields>> & { [key: string]: unknown })) = FormData<Fields>
```
altho is the value of the indexed object really of type unknown, or is it a set of known types, or an array/object of those known types?